syntax = "proto3";
package maum.brain.classification.cifar;

enum CifarClass{
    AIRPLANE = 0;
    AUTOMOBILE = 1;
    BIRD = 2;
    CAT = 3;
    DEER = 4;
    DOG = 5;
    FROG = 6;
    HORSE = 7;
    SHIP = 8;
    TRUCK = 9;
}

message Input{
    bytes image = 1;
}

message Output{
    CifarClass result = 1;
}

message SampleInput{
    bytes text = 1;
}

message SampleOutput{
    int32 result = 1;
}

service Classification{
    rpc GetTextResult(stream SampleInput) returns (SampleOutput);
    rpc GetResult(stream Input) returns (Output);
}

